scripts/o3de/o3de/cmake.py

```diff
@@ -158,7 +159,8 @@ def resolve_gem_dependency_paths(
     all_gems_json_data = manifest.get_gems_json_data_by_name(engine_path=engine_path, 
                                                              project_path=project_path, 
                                                              include_manifest_gems=True, 
-                                                             include_engine_gems=True)
+                                                             include_engine_gems=True,
+                                                             external_subdirectories=external_subdirs.split(';') if external_subdirs else None)
```

Comment:
A safer approach here is to split if the external_subdirs is an instance of str, otherwise assume it is a list already
```suggestion
                                                             external_subdirectories=external_subdirs.split(';') if isinstance(external_subdirs, str) else external_subdirs.)
```